elektrycznosc_statystyki_kk.sql

```diff
@@ -0,0 +1,356 @@
+
+/*
+ Zakres wskaźników:
+	Electricity production from coal sources (% of total)	EG.ELC.COAL.ZS
+	Electricity production from oil, gas and coal sources (% of total)	EG.ELC.FOSL.ZS
+	Electricity production from hydroelectric sources (% of total)	EG.ELC.HYRO.ZS
+	Electricity production from natural gas sources (% of total)	EG.ELC.NGAS.ZS
+	Electricity production from nuclear sources (% of total)	EG.ELC.NUCL.ZS
+	Electricity production from oil sources (% of total)	EG.ELC.PETR.ZS
+	Electricity production from renewable sources, excluding hydroelectric (kWh)	EG.ELC.RNWX.KH
+	Electricity production from renewable sources, excluding hydroelectric (% of total)	EG.ELC.RNWX.ZS
+*/
+--=============================================================================
+
+-- Sprawdzenie kto bierze udział w notowaniach
+select * from country c;
+
+-- Zauważam, że w rekordach ukrywają się zbiorcze statystyki World/ Europa itd.
+-- Można je zindentyfikować przez pole alpha2code, które zawiera wtedy cyfrę.
+-- Są też stowarzyszenia i frupy krajów:
+-- XC, EU, XE, XD, XR, XS, XJ, ZJ, XL XO, XM, XN, ZQ, XQ, XP, XU, OE,  ZG, ZF, XT
+-- niektóre dane w kWh inne w %
+
+--regiony
+select * from country c, 
+regexp_matches(alpha2code, '[0-9]');
+
+--kraje
+select * from country c
+where c.alpha2code !~ '[%0-9%]' 
+		and c.alpha2code !~'[X%]' 
+		and c.alpha2code not in ('EU', 'ZJ', 'ZQ', 'OE', 'ZG', 'ZF');
+
+
+--=================================================================================
+
+-- Wyciągam dla orientacji statystyki zbiorcze z rekordów dla regionów: World/ Europa itd.
+
+-- Średnie zużycie energi elekt. w regionach / świat
+select c.shortname as Region, 
+		round(avg(i.value)::numeric, 0) as zuzycie_regiony,
+		regexp_matches(alpha2code, '[0-9]')
+from indicators i
+join country c on i.countrycode = c.countrycode
+where lower(i.indicatorname) like '%electric power cons%'
+group by c.shortname, regexp_matches(alpha2code, '[0-9]')
+order by 1 desc;
+
+
+-- Sumaryczna produkcja energi elekt. w regionach / świat
+select c.shortname as Region, 
+		round(sum(i.value)::numeric, 0) as produkcja_regiony,
+		regexp_matches(alpha2code, '[0-9]')
+from indicators i
+join country c on i.countrycode = c.countrycode
+where lower(i.indicatorname) like '%electricity prod%'
+group by c.shortname, regexp_matches(alpha2code, '[0-9]')
+order by 1 desc;
+
+-- Sumaryczna produkcja energi elekt. w regionach / świat z podziałem na źródło
+
+select c.shortname as Region, 
+		i.indicatorname zrodlo,
+		round(sum(i.value)::numeric, 0) as produkcja_regiony_zrodlami,
+		regexp_matches(alpha2code, '[0-9]')
+from indicators i
+join country c on i.countrycode = c.countrycode
+where lower(i.indicatorname) like '%electricity prod%'
+group by c.shortname, i.indicatorname, regexp_matches(alpha2code, '[0-9]')
+order by 1 desc;
+
+--średnie zuzycie z podzialem na 10 lat
+select min(i."Year") from indicators i; --1960
+select max(i."Year") from indicators i; --2013
+
+drop table dziesiatki;
```

Comment:
W postgresie jest opcja IF EXISTS, z której warto korzystać. Tu to nie ma większego znaczenia, bo zapytania i tak odpalasz  sobie jedno po drugim, ale np. w sytuacji, gdy puszczasz jakiś jeden duży skrypt, którym cyklicznie odświeżasz dane w pewnej tabeli i dropujesz starą wersję, to fajnie jak to jest, bo bez tego warunku skrypt może wywrócić się np. w połowie.

Reply:
Jasne. Dzięki.